js/src/setup-mc/setup-stepper/setup-accounts/google-mc-account/getMatchingDomainAccount.js

```diff
@@ -6,9 +6,16 @@ import { getSetting } from '@woocommerce/settings'; // eslint-disable-line impor
 // See https://github.com/woocommerce/woocommerce-admin/issues/7781
 
 const getMatchingDomainAccount = ( existingAccounts = [] ) => {
-	const homeUrl = getSetting( 'homeUrl' );
-
-	return existingAccounts.find( ( el ) => el.domain === homeUrl );
+	/**
+	 * `homeUrl` has a trailing slash,
+	 * while `existingAccounts`'s `domain` has no trailing slash.
+	 * To be more defensive, we normalize the URLs with `new URL()` first
+	 * before doing the comparison.
+	 */
+	const homeUrl = new URL( getSetting( 'homeUrl' ) ).toString();
+	return existingAccounts.find(
+		( el ) => new URL( el.domain ).toString() === homeUrl
```

Comment:
💅 
I found sites could be verified with both `http` and `https` protocols. What about using `.hostname` to compare?

Reply:
@eason9487 , I wanted to treat `http` and `https` differently, because (for example) `http://www.example.com/` and `https://www.example.com/` could be entirely different sites. 

Here's a reference on Stack Exchange: [How to run two different websites in the same domain one with HTTP and the other with HTTPS?](https://serverfault.com/q/917933/59289)

Reply:
> I wanted to treat `http` and `https` differently

Got it.

---

Quoted from the Stack Exchange page:

> You can do this, but **please don't**.

> It's not worth adding as an extra answer, but I did this some years ago and now really regret it, as it is making it very hard to consolidate my entire site on https (this being modern best-practice). What everyone else is saying - that this is a bad idea - is true. 

> This would be technically possible but doesn't work at all in practice.

It's technically possible but I doubt that many websites will do this practically. 🤔 

No matter in either case, the chance of misjudgment is pretty low I guess, and even a misjudgment occurred, it will not prevent them from continuing to create a new account, so it's still a non-blocking comment.